docs/en/rules/Azure.Deployment.OuterSecret.md

```diff
@@ -0,0 +1,90 @@
+---
+reviewed: 2022-06-12
+severity: Critical
+pillar: Security
+category: Infrastructure provisioning
+resource: Deployment
+online version: https://azure.github.io/PSRule.Rules.Azure/en/rules/Azure.Deployment.OuterSecret/
+---
+
+# Secret value in deployment output
+
+## SYNOPSIS
+
+Do not use Outer deployments when references SecureString or SecureObject parameters.
+
+## DESCRIPTION
+
+Template child deployments can be scoped as either `outer` or `inner`. The `outer` scope uses parameters from the parent
+template which can be render into clear text. This exposes the secure information.
```

Comment:
```suggestion
Template child deployments can be scoped as either `outer` or `inner`.
When using `outer` scope evaluated deployments, parameters from the parent template are used directly within nested templates instead of enforcing `secureString` and `secureObject` types.

When passing secure values to nested deployments always use `inner` scope deployments to ensure secure values are not logging.

Bicep modules always use `inner` scope evaluted deployments.
```